[CoreML][fbcode] Add the `preprocess` python APIs

xta0 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #65076
* #65075
* #64523
* #64522
* __->__ #64521

Add the preprocess part for the coreml delegate. Check out the `example.py` for the usage.

Differential Revision: [D30585154](https://our.internmc.facebook.com/intern/diff/D30585154/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30585154/)!

xta0 committed:
[CoreML][fbcode] Add the `preprocess` python APIs

Add the preprocess part for the coreml delegate. Check out the `example.py` for the usage.

Differential Revision: [D30585154](https://our.internmc.facebook.com/intern/diff/D30585154/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30585154/)!

[ghstack-poisoned]

raziel reviewed (approved):
Some comments left in the internal diff.

xta0 committed:
Update on "[CoreML][fbcode] Add the `preprocess` python APIs"


Add the preprocess part for the coreml delegate. Check out the `example.py` for the usage.

Differential Revision: [D30585154](https://our.internmc.facebook.com/intern/diff/D30585154/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30585154/)!

[ghstack-poisoned]

xta0 committed:
Update on "[CoreML][fbcode] Add the `preprocess` python APIs"


Add the preprocess part for the coreml delegate. Check out the `example.py` for the usage.

Differential Revision: [D30585154](https://our.internmc.facebook.com/intern/diff/D30585154/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30585154/)!

[ghstack-poisoned]

xta0 committed:
Update on "[CoreML][fbcode] Add the `preprocess` python APIs"


Add the preprocess part for the coreml delegate. Check out the `example.py` for the usage.

Differential Revision: [D30585154](https://our.internmc.facebook.com/intern/diff/D30585154/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30585154/)!

[ghstack-poisoned]

xta0 committed:
Update on "[CoreML][fbcode] Add the `preprocess` python APIs"


Add the preprocess part for the coreml delegate. Check out the `example.py` for the usage.

Differential Revision: [D30585154](https://our.internmc.facebook.com/intern/diff/D30585154/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30585154/)!

[ghstack-poisoned]

xta0 committed:
Update on "[CoreML][fbcode] Add the `preprocess` python APIs"


Add the preprocess part for the coreml delegate. Check out the `example.py` for the usage.

Differential Revision: [D30585154](https://our.internmc.facebook.com/intern/diff/D30585154/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30585154/)!

[ghstack-poisoned]

xta0 committed:
Update on "[CoreML][fbcode] Add the `preprocess` python APIs"


Add the preprocess part for the coreml delegate. Check out the `example.py` for the usage.

Differential Revision: [D30585154](https://our.internmc.facebook.com/intern/diff/D30585154/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30585154/)!

[ghstack-poisoned]
